### ISSUE TITLE
[Feature]: Hope to optimize

### ISSUE THREAD

**Issue by designrans** [continue-revolution/sd-webui-animatediff#230]
### Expected behavior

When using video guidance for video style rendering, it is common to get stuck and the GPU usage is very high (4090). Can we optimize it.

**Comment by continue-revolution** [comment]
Do you mean ControlNet V2V? Can you give me more concrete context, like when the VRAM consumption is high? Do you get OOM before generation starts?

**Comment by designrans** [comment]
> Do you mean ControlNet V2V? Can you give me more concrete context, like when the VRAM consumption is high? Do you get OOM before generation starts?
<img width="677" alt="微信图片_20231017174932" src="https://github.com/continue-revolution/sd-webui-animatediff/assets/130264837/6502316a-8950-41e3-a76f-242be50433a7">
<img width="467" alt="微信图片_20231017174939" src="https://github.com/continue-revolution/sd-webui-animatediff/assets/130264837/af3708a0-599a-4710-ae06-e980ef1a5d5d">
<img width="667" alt="微信图片_20231017174943" src="https://github.com/continue-revolution/sd-webui-animatediff/assets/130264837/39b51286-8ea1-40f2-bcc3-ebc6fa786b95">
<img width="899" alt="微信图片_20231017175039" src="https://github.com/continue-revolution/sd-webui-animatediff/assets/130264837/55ac02b7-6753-48c7-a8a7-b6da015ce218">

**Comment by designrans** [comment]
3 seconds 480p, 24 frames of video, Controlnet is canny, 4090 takes 1 and a half hours. Did I set it incorrectly?

**Comment by continue-revolution** [comment]
If the frame number is 90, I don’t think it will OOM (cuda out of memory). I have tested locally with 95 frames, 512*522 You may want to use something like sdp or xformers.

If the frame number is ~300, one user reported that OOM will happen before generation starts (most likely because of CN). I will try to address this in #224 

If you set output to be 1024*1024, it is absolutely impossible to prevent OOM, unfortunately.

**Comment by min-star** [comment]
i have achieve the fluent video to video,but try it with sd_webui,it's huge slow.i don't know why,is the operator a mistake?i just use a lora with batch images with animatediff open.maybe comfyui's confgiuration is not suit for this extension
the cmdline arguments:python3 launch.py --no-gradio-queue --listen --port 12121 --opt-sdp-attention --share --skip-prepare-environment --enable-insecure-extension-access --api


![截图 2023-10-18 10-58-47](https://github.com/continue-revolution/sd-webui-animatediff/assets/59134868/4e563398-451b-4ebf-9359-74dc328f7c1e)

**Comment by continue-revolution** [comment]
@min-star  I‘ve not supported i2i batch. It will be available in v1.10.0 via #224

**Comment by min-star** [comment]
@continue-revolution thank you for you reply,thus the controlnet v2v is for txt2image,is the understand correct?

**Comment by continue-revolution** [comment]
For v1.9.2, yes. 

For v1.10.0 which will come later this week, not really. ControlNet v2v will be available for both t2i and i2i batch.

**Comment by min-star** [comment]
@okay,it's cool.

**Comment by davyzhang** [comment]
With 512x288 sized video, 26 frames can not get through using 16GB GPURam instance provided by colab.

```
2023-10-24 09:27:53,838 - ControlNet - INFO - Loading model: control_v11p_sd15_openpose [cab727d4]
2023-10-24 09:28:02,099 - ControlNet - INFO - Loaded state_dict from [/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_openpose.pth]
2023-10-24 09:28:02,099 - ControlNet - INFO - controlnet_default_config
2023-10-24 09:28:07,696 - ControlNet - INFO - ControlNet model control_v11p_sd15_openpose [cab727d4] loaded.
2023-10-24 09:28:08,027 - ControlNet - INFO - Loading preprocessor: dw_openpose_full
2023-10-24 09:28:08,027 - ControlNet - INFO - preprocessor resolution = 512
2023-10-24 09:31:07,753 - ControlNet - INFO - ControlNet Hooked - Time = 194.59531688690186
100% 20/20 [01:14<00:00,  3.74s/it]
^C
```

**Comment by LIQUIDMIND111** [comment]
NOTE - I have an rtx 2060 6 VRAM GPU and i can make 512x768 animations at 8fps and 124 max frames and get great results.... but if i use an 1024x576 model - it will OOM and crash. I can make an animation of 512x512 of 64 frames max in les tan 10 minutes using SDP and reducing BATCH CONTEXT to 8.... in MY case, that works a charm!

**Comment by continue-revolution** [comment]
V2V VRAM problem should have been resolved in v1.11.1
